python/ray/WebUI.ipynb

```diff
@@ -1,61 +1,194 @@
+    "                            num_tasks_box.value = new_high / 100. * num_tasks\n",
+    "                    else:\n",
+    "                        start_box.value = round(diff * low, 2)\n",
+    "                        end_box.value = round(diff * high, 2)\n",
+    "                elif event['owner'] == num_tasks_box:\n",
+    "                    if num_tasks_box.value > 0:\n",
+    "                        range_slider.value = (0, int(100 * float(num_tasks_box.value) / num_tasks))\n",
+    "                    elif num_tasks_box.value < 0:\n",
+    "                        range_slider.value = (100 + int(100 * float(num_tasks_box.value) / num_tasks), 100)\n",
+    "                else:\n",
+    "                    raise ValueError('Unknown event owner!')\n",
+    "\n",
+    "                if not update:\n",
+    "                    return\n",
     "\n",
-    "def handle_submit(sender):\n",
-    "    pp = pprint.PrettyPrinter()\n",
-    "    pp.pprint(ray.global_state.object_table(object_search.value))\n",
+    "                diff = largest - smallest\n",
+    "                low, high = map(lambda x: x / 100., range_slider.value)\n",
     "\n",
-    "object_search.on_submit(handle_submit)"
+    "                if breakdown_opt.value == total_time_value:\n",
+    "                    tasks = ray.global_state.task_profiles(start=smallest + diff * low, end=smallest + diff * high)\n",
+    "                elif breakdown_opt.value == total_tasks_value:\n",
+    "                    if range_slider.value[0] == 0:\n",
+    "                        tasks = ray.global_state.task_profiles(num_slice=int(num_tasks * high), fwd=True)\n",
+    "                    else:\n",
+    "                        tasks = ray.global_state.task_profiles(num_slice=int(num_tasks * (high - low)), fwd=False)\n",
+    "                else:\n",
+    "                    raise ValueError('Value \"{}\" is not a legal breakdown value'.format(breakdown_opt.value))\n",
+    "\n",
+    "                update(smallest, largest, num_tasks, tasks)\n",
+    "       \n",
+    "    range_slider.observe(update_wrapper, names='value')\n",
+    "    breakdown_opt.observe(update_wrapper, names='value')\n",
+    "    start_box.observe(update_wrapper, names='value')\n",
+    "    end_box.observe(update_wrapper, names='value')\n",
+    "    num_tasks_box.observe(update_wrapper, names='value')\n",
+    "   \n",
+    "    update_wrapper(INIT_EVENT)\n",
+    "   \n",
+    "    display(start_box, end_box, range_slider, num_tasks_box, breakdown_opt)\n",
+    "   \n",
+    "    return start_box, end_box, range_slider, breakdown_opt"
    ]
   },
   {
-   "cell_type": "markdown",
-   "metadata": {},
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
    "source": [
-    "**Evaluate the box below to search for tasks.**"
+    "earliest, latest, num = ray.global_state.job_length()\n",
```

Comment:
My intended use case was to give a clear understanding of what the possible time windows are for when a user would use the start and end time window boxes.